src/jsonpatcherproxy.js

```diff
@@ -113,25 +113,42 @@ const JSONPatcherProxy = (function() {
       }
     }
     // let's start with this operation, and may or may not update it later
+    const valueBeforeReflection = tree[key];
+    const wasKeyInTreeBeforeReflection = tree.hasOwnProperty(key);
+    if (isTreeAnArray && !isNonSerializableArrayProperty) {
+      const index = parseInt(key, 10);
+      if (index > tree.length) {
+        trapForSet(instance, tree, (index - 1) + '', undefined);
```

Comment:
Could you add a code comment that this calls set trap for automatically added `undefined`->`null` elements of an array?

Reply:
Done